addons/project/models/project_task.py

```diff
@@ -723,7 +723,7 @@ def _compute_display_follow_button(self):
     def _get_group_pattern(self):
         return {
             'tags_and_users': r'\s([#@]%s[^\s]+)',
-            'priority': r'\s(!)',
+            'priority': r'(?:^|\s)(!{1,3})(?=\s|$)',
```

Comment:
why not simply having:

```suggestion
            'priority': r'\s(!{1,3})',
```

I don't understand why the regex is now complex, TBH. 🤔

Reply:
> why not simply having:
> 
> I don't understand why the regex is now complex, TBH. 🤔

When the user inputs a task name such as `task !a @marc !!!`, the system creates a one-star task instead of a three-star task
```py
pattern1 = r'\s(!{1,3})'
pattern2 = r'(?:^|\s)(!{1,3})(?=\s|$)'

task_name = 'task !a @marc !!!'

re.findall(pattern1, task_name) -> output: ['!']
re.findall(pattern2, task_name) -> output: ['!', '!!!']
```

what do you think ?

Reply:
okay then, we could have something like: `r'\s(!{1,3})(?=\s|$)'`, no?